### ISSUE TITLE
Adding dynamic data to angular v12 atcb control - Rebind event

### ISSUE THREAD

**Issue by mshaheerimran** [jekuer/add-to-calendar-button#25]
The atcb div does not re-render through *ngIf clause or any other way. The actual requirement is to rebind the json data object as the data that is to be passed to the atcb widget is not loaded as soon as the html is rendered. It is received as part of a response from an Ajax call. By that time if we change the model variable which is bound to the json object, it does not reflect. We tried with an *ngIf clause and only show the atcb div/control after the API response has loaded but in this case the button does not render at all. 

Is there any way to rebind the control's json variable or make changes to the json variable after the control has rendered. 

@jekuer

**Comment by jekuer** [comment]
Hi,
Not sure if I got it 100% right, but if you want to dynamically create the buttons, you can simply show the whole div in the respective case and then re-run atcb_init(). It would then find all new buttons and render them accordingly.
So, if you receive the JSON asynch, I would only set the div as soon as you have the JSON. And run atcb_init() after the JSON has been received and the div placed,
Does this solve your issue?

**Comment by jekuer** [comment]
Another solution would be to include some kind of asynch watcher/listener to the button, but this feels a little bit off and unnecessary at the moment.

**Comment by mshaheerimran** [comment]
HI @jekuer , 

Actually the issue was resolved somehow, and I wasn't able to update back here with what resolved the issue.

Your suggestion about rendering the div only when the async data is returned back was something that we were doing already but still the json was not being update back to the atcb control. What made it work was setting the json value in a timeout block (I assume, the scope variable was not refreshed all the way up to the atcb directive and it was still reading the old variable value in the json instead of the one received from the async call). After setting the async data in the timeout() block, it  forced the value to be rebound to the atcb control. 

Not sure if this can be further optimized or what is the root cause of the variable value not being updated back to atcb control, this is how we resolved it. Let me know what you think on this and if this can be further improved.
